apps/website/src/components/ComponentIcon.astro

```diff
@@ -0,0 +1,102 @@
+---
+import { renderJSX } from 'astro/dist/runtime/server';
+import fs from 'node:fs';
+
+export type Props = {
+  /** name of the file to read code example from, should be located in ~/examples folder */
+  icon: string;
+};
+
+const { icon } = Astro.props;
+
+// baseUrl refers to project root
+const baseUrl = import.meta.env.PROD
+  ? new URL('..', import.meta.url).href
+  : new URL('../../', import.meta.url);
+
+let code = '';
+
+if (icon) {
+  const codePath = new URL(`./src/icons/component-icons/${icon}.astro`, baseUrl);
+  code = await fs.promises.readFile(codePath, 'utf8');
+}
+
+// const componentName = icon.split('.')[0];
+// const capitalize = (s: string) => s.charAt(0).toUpperCase() + s.slice(1);
+---
+
+<div class='thumbnail' set:html={code} />
+
+<style lang='scss' is:global>
+  .thumbnail svg {
+    height: calc(var(--space-6) * 7);
+    width: 100%;
+  }
+  .main-color {
+    fill: #fff;
```

Comment:
contrast is pretty bad in a few places
![](https://user-images.githubusercontent.com/9084735/206533817-f6e441cc-96b4-4c44-9676-9c04d8327673.png)

Reply:
Changed color a bit